### PR TITLE
Add query command

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Added query command

Example output:

Not-json:
```
Query result:
          QueryResult  {"input":"hi","other":"yoyo"}
```

JSON:
```
{
  "queryResult": [
    {
      "input": "hi",
      "other": "yoyo"
    }
  ]
}
```

## Why?
Part of rewrite effort

## Checklist

1. Closes 

2. How was this tested:
Added test

3. Any docs updates needed?

### CODE REVIEW

File: temporalcli/commandsmd/commands.md [hunk 1]
```diff
@@ -345,7 +345,27 @@ Use the command options below to change the information returned by this command
...
-TODO
+The `temporal workflow query` command is used to [Query](/concepts/what-is-a-query) a
+[Workflow Execution](/concepts/what-is-a-workflow-execution)
+by [ID](/concepts/what-is-a-workflow-id).
+
+```
+temporal workflow query \
+		--workflow-id MyWorkflowId \
+		--name MyQuery \
+		--input '{"Input": "As-JSON"}'
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply temporalcli/commandsmd/commands.md:356]: It's just duping what signal also shows, but yeah we could change it

[the remark this answers is not in the document]
Comment (Sushisource) [reply temporalcli/commandsmd/commands.md]: I don't think we need to agonize. It's pretty clear for anyone who's ever used a CLI to submit JSON what's going on.

**Review by dandavison — APPROVED** [review]
LGTM!

File: temporalcli/commandsmd/commands.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cretz) [reply temporalcli/commandsmd/commands.md]: FWIW, the signal one was copied over from previous CLI. I figure we;ll do a real doc review/reworking at a later time.

File: temporalcli/commandsmd/commands.md [hunk 2]
```diff
@@ -345,7 +345,27 @@ Use the command options below to change the information returned by this command
...
+```
+
+Use the options listed below to change the command's behavior.
+
+#### Options
+
+* `--name` (string) - Query Name. Required.
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply temporalcli/commandsmd/commands.md:363]: Yeah it is "type" in the old one. "name" is consistent with the signal command, now. I actually prefer type generally, though. So maybe in an ideal world we'd change signal instead. I'm fine either way, but it'd be kinda nice to have them all be consistent.

File: temporalcli/commandsmd/commands.md [hunk 3]
```diff
@@ -345,7 +345,27 @@ Use the command options below to change the information returned by this command
...
+* `--reject-condition` (string) - Optional flag for rejecting Queries based on Workflow state. Valid
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply temporalcli/commandsmd/commands.md:364]: Thanks, I wanted that but somehow I didn't see the existing ones

File: temporalcli/commandsmd/commands.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cretz) [reply temporalcli/commandsmd/commands.md:363]: Concur about consistency, but when hitting these crossroads in the CLI rewrite today, we've erred towards not breaking compat (especially on commonly used commands like this). I can be convinced either way, but I think I prefer `--type` just for compat.

[the remark this answers is not in the document]
Comment (dandavison) [reply temporalcli/commandsmd/commands.md]: In the current CLI, signal and update use `--name` whereas query uses `--type`.  However, IMO we need the flag name to be the same for signal, query, and update, so a backwards compatibility break with our 0.x `temporal` is inevitable.

**Pull request merged**